index.rst

```diff
@@ -338,4 +338,121 @@ smaller than a single alert message.
+1 TiB of ephemeral storage.  When initiating Kafka and the alert stream component containers, we made sure
+to constrain the Kafka container to its own node and separate the other containers to different nodes so as
+not to have Kafka competing with other containers for resources.  With these settings, the lagged producer
+problem is resolved.  In practice for the expected LSST alert distribution system in production, this will
+likely not be a problem since there should be one producer per CCD, with the planned compute parallelization
+of the pipeline.
+
+Results
+^^^^^^^
+For scale testing of consumers, we used 1 consumer printing every 100th alert + 1 fast consumer dropping alerts,
+3+1 consumers, 5+1, and 10+1, testing both with postage stamps and without.
+
+Below shows the memory utilization of Kafka with an increased number of consumers and larger allotted
+instances on AWS.
+
+.. figure:: _static/memoryConsumers.png
+   :width: 55%
+   :align: center
+   :name: figure-14
+
+   Kafka container memory utilization against number of consumers.
+
+For the 1+1 consumer test, instances were only allotted 32 GB of memory.  The other tests had 256 GiB memory
+available and utilize more memory.  Tests with stamps included in alerts are significantly more memory
+intensive for Kafka than tests without including postage stamps cutouts.
+
+For these tests there is no difference for the network traffic into Kafka, as we are still sending 10,000 alerts
+in each visit burst.  The network traffic out of Kafka increases linearly with more consumers, as expected,
+up to 180000 KiBps (180 Mbps) for 10+1 consumers, since each consumer pulls its own stream.
+
+.. figure:: _static/netOutConsumers.png
+   :width: 55%
+   :align: center
+   :name: figure-15
+
+   Network traffic out of Kafka against number of consumers.
+
+Below shows the length of time it takes for producers to serialize alerts into Avro and produce to the Kafka
+broker.  Having more consumers has a small impact on the producer time.
+For the 1+1 consumer test, we used only one producer sending 10,000 alerts (unparallelized), which is
+why the time is slightly higher.  For the other three tests, the serialization and producing time increases
+slightly and linearly to 9 seconds.  With more producers producing in parallel (one per CCD),
+this number would likely be lower.
+
+.. figure:: _static/serialTimeConsumers.png
+   :width: 55%
+   :align: center
+   :name: figure-16
+
+   Serialization and Kafka submission time for producers against number of consumers.
+
+The transit time of alerts, the time it takes for alerts to be read by the consumers after they have been
+submitted into the Kafka queue, increases with number of consumers.  Below shows the relationship between
+the transit time and number of consumers.
+
+
+.. figure:: _static/transitTimeConsumers.png
+   :width: 55%
+   :align: center
+   :name: figure-17
+
+   Transit time of alerts to consumers against number of consumers.
```

Comment:
I don't know about the first jump.  I plotted this also as total network stream size out of Kafka vs transit time, and that jump is at 3 GB.  I'm not sure if that is a special number anywhere.  For the last huge jump there, any transit time larger than 39 seconds will just balloon with the length of time the experiment runs (because it cannot catch up with the alerts and falls farther and farther behind.)  It would be smaller (down to 39 seconds) if I did a shorter experiment (and bigger with a longer one), since it's just an average.  I am not sure why it takes longer than 39 seconds.  I checked the disk IO read speed for the instance and the instance to instance networking for 11 consumers worth of streams, and neither of those are bottlenecks.